src/services/assets.js

```diff
@@ -343,28 +355,59 @@ export function getExchangeRates(assets: string[]): Promise<?Object> {
     return token;
   });
 
-  return cryptocompare.priceMulti(assets, targetCurrencies)
+  return cryptocompare
+    .priceMulti(assets, targetCurrencies)
     .then(data => {
       // rename HOLO to HOT
       if (data[HOLO]) {
         data[HOT] = { ...data[HOLO] };
         delete data[HOLO];
       }
-      /**
-       * sometimes symbols have different symbol case and mismatch
-       * between our back-end and crypto compare returned result
-       */
-      return Object.keys(data).reduce((mappedData, returnedSymbol) => {
-        const walletSupportedSymbol = assets.find((symbol) => isCaseInsensitiveMatch(symbol, returnedSymbol));
-        if (walletSupportedSymbol && !mappedData[walletSupportedSymbol]) {
-          mappedData = {
-            ...mappedData,
-            [walletSupportedSymbol]: data[returnedSymbol],
-          };
-        }
-        return mappedData;
-      }, {});
-    }).catch(() => ({}));
+      return data;
+    })
+    .catch(() => {
+      reportErrorLog('getLegacyExchangeRates cryptocompare.cryptocompare failed', { assets });
+      return null;
+    });
+}
+
+export async function getExchangeRates(assetSymbols: string[]): Promise<?Object> {
+  if (isEmpty(assetSymbols)) {
+    reportLog('getExchangeRates received empty assetSymbols array', { assetSymbols });
+    return null;
+  }
+
+  // CryptoCompare is legacy price oracle, however, the change to new one is feature flagged
+  const useLegacyCryptoCompare = firebaseRemoteConfig.getBoolean(FEATURE_FLAGS.USE_LEGACY_CRYPTOCOMPARE_TOKEN_PRICES);
+
+  let rates = useLegacyCryptoCompare
+    ? await getLegacyExchangeRates(assetSymbols)
+    : await getCoinGeckoTokenPrices(assetSymbols);
+
+  // by any mean if CoinGecko failed let's try legacy way
+  if (!useLegacyCryptoCompare && isEmpty(rates)) {
+    rates = await getLegacyExchangeRates(assetSymbols);
+  }
+
+  if (!rates) {
+    reportErrorLog('getExchangeRates failed: no rates data', { rates });
```

Comment:
Maybe here it would be good to also have a method used to get the rates (legacy/non-legacy) as extra data?

Reply:
👍